changelog/AliasAssign.md

```diff
@@ -0,0 +1,53 @@
+# Add Alias Assignment
+
+This adds the ability for an alias declaration inside a template to be
+assigned a new value. For example, the recursive template:
+
+---
+template staticMap(F, T...)
+{
+    static if (T.length == 0)
+	alias staticMap = AliasSym!();
+    else
+        alias staticMap = AliasSym!(F!(T[0]), staticMap!(T[0 .. T.length]));
+}
+---
+
+can now be reworked into an iterative template:
+
+---
+template staticMap(F, T...)
+{
+    alias A = AliasSeq!();
+    static foreach (t; T)
+	A = AliasSeq!(A, F!t); // alias assignment here
+    alias staticMap = A;
+}
+---
+
+Using the iterative approach will eliminate the combinatorial explosion of recursive
+template instantiations, eliminating the associated high memory and runtime costs,
+as well as eliminating the issues with limits on the nesting depth of templates.
+It will eliminate the obtuse error messages generated when deep in recursion.
+
+The grammar:
+
+---
+AliasAssign:
+    Identifier = Type;
```

Comment:
Why is reassignment limited to types? What about aliases for other symbols?

Reply:
Because if this doesn't work, it isn't worth the effort to extend it.